### ISSUE TITLE
`id-match` requires separate config for property name

### ISSUE THREAD

**Issue by tinymins** [eslint/eslint#11797]
**What rule do you want to change?**
id-match

**Does this change cause the rule to produce more or fewer warnings?**
Yes

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
/* eslint id-match: [2, "^[^_][a-zA-Z]*$", { 
  "propertiesPattern": "^[a-z_][a-zA-Z]*$",
  "errorMessage": "Identifier '{{name}}' is not in camelcase.",
  "propertiesErrorMessage": "Properties identifier '{{name}}' is not in lower camelcase."
}] */
// bad
const _A = { a: 1 };
const _A = { _a: 1 };
const A = { a_b: 1 };
const A = { B: 1 };

// good
const A = { a: 1 };
const A = { _a: 1 };
const A = { aB: 1 };
const A = { b: 1 };
```

**What does the rule currently do for this code?**
Cannot separate var name with prop name, sometimes we want to allow underscore in prop names but not in normal var names.

**What will the rule do after it's changed?**
Additional config added, no affect for prev version.

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by tinymins** [comment]
https://github.com/eslint/eslint/pull/10477

**Comment by g-plane** [comment]
Hey @tinymins , thanks for the issue.

Alternatively, you can consider [camelcase](https://eslint.org/docs/rules/camelcase#require-camelcase-camelcase) and set `"properties"` false.

**Comment by tinymins** [comment]
> Hey @tinymins , thanks for the issue.
> 
> Alternatively, you can consider [camelcase](https://eslint.org/docs/rules/camelcase#require-camelcase-camelcase) and set `"properties"` false.

Hi, @g-plane https://github.com/eslint/eslint/issues/10473
